### PR TITLE
[4.0] Multilingual status: adding more infos and clarifying some

### PR DESCRIPTION
**Pull request by infograf768**
### Summary of Changes
This PR adds a Notice and displays the language concerned when the Content Language for that language has been deleted and remains in the site an orphan home page while the site language is published.
It also adds a Notice when a Content Language is in the trash.

It also reinstate all states of a content language to display the correct icons.
published = 1
publsihed = 0
published = -2 (i.e. trash) and for this one displays the trash icon

### Testing Instructions
Install a multilingual site with all languages available to install
French fr-FR
German de-DE
Persian fa-IR

Then use the Cpanel Multilingual Sample Data module to create a basic.

First test:
Display the Content languages Manager
Trash AND delete the fa-IR Persian content language.

You should get this when displaying the multilingual status module

screen shot 2019-03-02 at 18 25 09

Then unpublish the French fr-FR Content Language.
And trash (do NOT delete) the German de-DE Content language.
I.e. Manager should look like this
screen shot 2019-03-02 at 18 28 10

You should get this when displaying the multilingual status module
screen shot 2019-03-02 at 18 44 18

### Edit
Unpublish also the fr-FR Home page to get a new notice.
Deleted Content Language error now displays the notice as last.
see https://github.com/joomla/joomla-cms/pull/24062#issuecomment-469004768

### Reminder
It is OK to have an Unpublished Content Language (as French is above) as it lets the site owner prepare a new Language for the multilingual site.

### CODE REVIEW

File: administrator/components/com_languages/tmpl/multilangstatus/default.php [hunk 1]
```diff
@@ -209,11 +241,21 @@
 						</td>
 				<?php endif; ?>
 				<?php // Published Content languages ?>
-				<?php if ($status->lang_code && $status->published) : ?>
+				<?php if ($status->lang_code && $status->published == 1) : ?>
```
Comment (Quy) [inline administrator/components/com_languages/tmpl/multilangstatus/default.php]: Change it similar to lines 289 - 300 where all the if statements are inside one `<td class="text-center">` and `</td>`.

Comment (infograf768) [reply administrator/components/com_languages/tmpl/multilangstatus/default.php]: Makes sense. Will do.
Just found out some issues I have to solve also. So this is still WIP.

**Comment by infograf768** [comment]
@Quy 
Modification done.
Added new Notice when both Content Language and Home page are unpublished.

We should now get in that case:

screen shot 2019-03-03 at 10 22 15

can be fully tested now.

**Comment by Quy** [comment]
> Added new Notice when both Content Language and Home page are unpublished.

The `Published Default Home pages` has all checks even though all pages are unpublished.

File: administrator/components/com_languages/tmpl/multilangstatus/default.php [hunk 2]
```diff
@@ -11,9 +11,14 @@
 
 use Joomla\CMS\Language\Text;
 
-$notice_homes     = $this->homes == 2 || $this->homes == 1 || $this->homes - 1 != count($this->contentlangs) && ($this->language_filter || $this->switchers != 0);
-$notice_disabled  = !$this->language_filter	&& ($this->homes > 1 || $this->switchers != 0);
-$notice_switchers = !$this->switchers && ($this->homes > 1 || $this->language_filter);
+$notice_homes           = $this->homes == 2 || $this->homes == 1 || $this->homes - 1 != count($this->contentlangs) && ($this->language_filter || $this->switchers != 0) && count($this->site_langs) == count($this->contentlangs);
+$notice_disabled        = !$this->language_filter && ($this->homes > 1 || $this->switchers != 0);
+$notice_switchers       = !$this->switchers && ($this->homes > 1 || $this->language_filter);
```
Comment (Quy) [inline administrator/components/com_languages/tmpl/multilangstatus/default.php]: Remove extra spaces before `=`.

File: administrator/components/com_languages/tmpl/multilangstatus/default.php [hunk 3]
```diff
@@ -51,6 +56,34 @@
...
+					</td>
+					<td>
+						<?php echo Text::sprintf('COM_LANGUAGES_MULTILANGSTATUS_CONTENT_LANGUAGE_TRASHED', $status->lang_code); ?>
+					</td>
+				</tr>
+			<?php endif; ?>
+		<?php endforeach; ?>
+		<?php // Displays error when both Content Language and Home page are unpublished ?>
+		<?php foreach ($this->statuses as $status) : ?>
+			<?php if ($status->lang_code && $status->published == 0 && !$status->home_language) : ?>
```
Comment (Quy) [inline administrator/components/com_languages/tmpl/multilangstatus/default.php]: Can this section be combined with the above loop?

File: administrator/components/com_languages/tmpl/multilangstatus/default.php (hunk 2 above)
Comment (infograf768) [reply administrator/components/com_languages/tmpl/multilangstatus/default.php]: will do

File: administrator/components/com_languages/tmpl/multilangstatus/default.php (hunk 3 above)
Comment (infograf768) [reply administrator/components/com_languages/tmpl/multilangstatus/default.php]: I guess so. On it.

**Comment by infograf768** [comment]
>> Added new Notice when both Content Language and Home page are unpublished.

> The Published Default Home pages has all checks even though all pages are unpublished.

I do not understand this comment.

**Comment by Quy** [comment]
I don’t get the same result in the last column as in your last screenshot.

**Comment by infograf768** [comment]
> I don’t get the same result in your last screenshot.

Please explain with more details. 
The exact settings you have for each language and the resulting screenshot.

File: administrator/components/com_languages/tmpl/multilangstatus/default.php (hunk 2 above)
Comment (Quy) [reply administrator/components/com_languages/tmpl/multilangstatus/default.php]: Do the same for above.

**Comment by Quy** [comment]
I am no longer at my computer so it will have to wait until tomorrow. I followed all your testing instructions. Only the last screenshot that I did not get the same result. All pages are unpublished.

File: administrator/components/com_languages/tmpl/multilangstatus/default.php (hunk 2 above)
Comment (infograf768) [reply administrator/components/com_languages/tmpl/multilangstatus/default.php]: done

**Comment by infograf768** [comment]
I may have forgotten to ask testing by unpublishing the French Home page AND its content language, in order to get the Notice and the right icons.
The order of the Notices may also have  changed. One has to test multiple cases.

**Comment by infograf768** [comment]
We may need though another Notice when site language is enabled, content language is published and home page is unpublished.

I.e. this case for French and German
screen shot 2019-03-04 at 08 41 19


On it.

**Comment by infograf768** [comment]
Added notice when both content language and home page missing. added trash icon when home page is in the trash.

We now get in these cases:

screen shot 2019-03-04 at 12 40 18

**Comment by infograf768** [comment]
Added warning icon when home page does not exist.
screen shot 2019-03-04 at 12 56 37

**Comment by infograf768** [comment]
@alikon

File: administrator/components/com_languages/Helper/MultilangstatusHelper.php [hunk 4]
```diff
@@ -101,8 +101,8 @@ public static function getStatus()
 
 		// Select the language home pages.
 		$query->select('l.home AS home')
-			->select('l.language AS home_language')
-			->join('LEFT', '#__menu AS l ON l.language = a.lang_code AND l.home=1 AND l.published=1 AND l.language <> \'*\'')
+			->select('l.published AS home_published')
+			->join('LEFT', '#__menu AS l ON l.language = a.lang_code AND l.home=1 AND l.language <> \'*\'')
```
Comment (alikon) [inline administrator/components/com_languages/Helper/MultilangstatusHelper.php]: please use `quoteName()`

**Review by alikon — CHANGES_REQUESTED** [review]

File: administrator/components/com_languages/Helper/MultilangstatusHelper.php (hunk 4 above)
Comment (infograf768) [reply administrator/components/com_languages/Helper/MultilangstatusHelper.php]: There are many places in this helper where we should use `quoteName()`
Shall we not reserve this for another specific patch ?

Comment (infograf768) [reply administrator/components/com_languages/Helper/MultilangstatusHelper.php]: and also `$db->quote()`

**Comment by Quy** [comment]
Please fix conflicts.

**Comment by infograf768** [comment]
@Quy 
This needs a lot of work to solve the conflicts and also update the queries.
I may have to create a new PR
Will try when I have time.

**Comment by infograf768** [comment]
I am afraid this branch is dead. Please test this new PR
https://github.com/joomla/joomla-cms/pull/27184

Closing

**Pull request closed without merging**